### PR TITLE
Compartmentalize spew more

### PR DESCRIPTION
**Pull request by thockin**
xref #103942

/kind cleanup

```release-note
NONE
```

### CODE REVIEW

**Comment by thockin** [comment]
Had to allow importing apimachinery from code-generator

File: staging/src/k8s.io/endpointslice/util/controller_utils.go [hunk 1]
```diff
@@ -303,13 +303,7 @@ func stringPtrChanged(ptr1, ptr2 *string) bool {
 // copied from k8s.io/kubernetes/pkg/util/hash
 func deepHashObject(hasher hash.Hash, objectToWrite interface{}) {
 	hasher.Reset()
-	printer := spew.ConfigState{
-		Indent:         " ",
-		SortKeys:       true,
-		DisableMethods: true,
-		SpewKeys:       true,
-	}
```
Comment (aojea) [inline staging/src/k8s.io/endpointslice/util/controller_utils.go:311]: the new printer has two additional options but  seems ok at first sight

https://github.com/kubernetes/kubernetes/blob/a28f14089cfa47ef9c57f9f283e1504a68f616d6/staging/src/k8s.io/apimachinery/pkg/util/dump/dump.go#L30-L49

Comment (thockin) [reply staging/src/k8s.io/endpointslice/util/controller_utils.go:311]: Yeah, these are not saved anywhere, as far as I can tell.

**Pull request merged**